### ISSUE TITLE
Clear icons when clearing private data

### ISSUE THREAD

**Issue by pocmo** [mozilla-mobile/fenix#12980]
See https://github.com/mozilla-mobile/android-components/issues/7766 for details.

We added `BrowserIcons.clear()` to make it possible to remove all icons from disk and memory.

**Comment by BranescuMihai** [comment]
Question for UX: From a user's perspective, which of the two options from the `Delete browsing data` fragment you would associate with clearing the favicons:
- Browsing history and site data
- Cached images and files

I would personally go with the first, since that is the one that I associate with removing all trace of me visiting a website.

**Comment by apbitner** [comment]
Assigning @AmyYLee for UX follow-up

**Comment by AmyYLee** [comment]
> Question for UX: From a user's perspective, which of the two options from the `Delete browsing data` fragment you would associate with clearing the favicons:
> 
>     * Browsing history and site data
> 
>     * Cached images and files
> 
> 
> I would personally go with the first, since that is the one that I associate with removing all trace of me visiting a website.

@betsymi on this one since it's about strings. I personally think of the second option because I think of favs as part of cached images.

**Comment by betsymi** [comment]
I do love a good string debate! I am going to recommend **Browsing history and site data** because I find the language simpler and easier to understand. "Cached" veers on the side of jargony. While it may be more specific, I'd rather veer on the side of removing all traces.

**Comment by Mugurell** [comment]
Hey @betsymi, @AmyYLee , just to confirm, because it came up in review,
Clearing the icons should happen when the user selects
- "Browsing history and site data"
or
- "Cached images and files"

?

**Comment by betsymi** [comment]
Browsing history and site data

**Comment by softvision-miralobontiu** [comment]
Hi all,
Verified as fixed on a debug build from 9/7 (AC: 58.0.20200906130403, b6c304b2a) on Samsung Galaxy Tab S6 (Android 9).
In the screenshots below, the first image represents the mozac_browser_icons before the deletion, and the second screenshot is after deletion. I've only selected to delete "Browsing history and site data" from the "Delete browsing data" page.

![deletebrowsingdata](https://user-images.githubusercontent.com/40694158/92375674-1c00c180-f10a-11ea-8c42-3fcc650eb051.png)
